Increase golden test sampleSize from 4 to 8

Anviking opened the pull request:
# Overview

- [ ] To cover more cases of complex ADTs


# Comments

Let's wait with merging this until we have fewer PRs in progress, and re-run:
```
rm test/data/Cardano/Wallet/Api/*.json
rm test/data/Cardano/Wallet/Api/*.json
stack test cardano-wallet
```

Also, all in one commit, filter by file extension to find the tiny `.hs` stuff

<!-- 
Don't forget to:

 ✓ Self-review your changes to make sure nothing unexpected slipped through
 ✓ Assign yourself to the PR
 ✓ Assign one or several reviewer(s)
 ✓ Once created, link this PR to its corresponding ticket
 ✓ Acknowledge any changes required to the Wiki
-->

KtorZ commented:
Right, but at least, we don't spread the usage of `V1` to the whole codebase anymore. So that's fine.

Anviking committed:
Increase golden sampleSize from 4 to 8 (and regenerate)

I ran
```
rm test/data/Cardano/Wallet/Api/*.json
rm test/data/Cardano/Wallet/Api/*.json
stack test cardano-wallet
```
